### DISCUSSION TITLE
Pyraf can be installed on Windows but not FCNTL and Termios

### DISCUSSION THREAD

**Discussion by LudmilaSchnei** [iraf-community/iraf#425] [General]
Hello,
I successfully installed pyraf on my windows via pip.
Unfortunately to work pyraf needs other modules like fcntl and termios, which are both not available for windows.
There might be more modules like this that are needed but that cannot be installed on a windows system.
Do you know of a solution to make pyraf run on windows?

**Comment by olebole** [comment]
I am moving this to the discussion forum as this is not a bug.

**Comment by olebole** [comment]
PyRAF requires IRAF to work, and IRAF is not available under native Windows.

You can, however, install IRAF (and PyRAF) under the ["Windows subsystem for Linux" (WSL)](https://learn.microsoft.com/en-us/windows/wsl/about). The simplest way there would be to install a Debian or Ubuntu distribution under WSL, and then install the required packages from the distribution ([iraf](https://packages.debian.org/stable/iraf) and [python3-pyraf](https://packages.debian.org/stable/python3-pyraf), maybe other IRAF packages like [iraf-st4gem](https://packages.debian.org/trixie/iraf-st4gem) or [iraf-fitsutil](https://packages.debian.org/stable/iraf-fitsutil), depending on your needs).
